spring-session/src/main/java/org/springframework/session/data/mongo/StandardDBObjectToMongoSessionConverter.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2002-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.session.data.mongo;
+
+import com.mongodb.DBObject;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.session.Session;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.util.Date;
+import java.util.Map;
+
+/**
+ * {@code Converter} implementation which returns session deserialized from a BSON object
+ *
+ * @author Jakub Kubrynski
+ * @since 1.2
+ */
+class StandardDBObjectToMongoSessionConverter implements Converter<DBObject, MongoExpiringSession> {
```

Comment:
I'm thinking rather than injecting two `Converter` implementations we should require a `GeneircConverter`. This would allow for a few benefits:

1) It would allow the same class to serialize and deserialize.
2) It would allow for converting other types (if necessary) in the future without adding more setter methods.
3) Likely we could experiment with registering the default Spring `GeneircConverter` so any hooks for automatic conversion are included with this feature.